.docker/Dockerfile-test-hsm

```
@@ -0,0 +1,32 @@
+FROM golang:1.23-alpine3.21 AS builder
+
+RUN apk -U --no-cache --upgrade --latest add build-base git gcc bash
```

You shouldn't need `-U` as you don't care about cache? I've not encountered an need for `--upgrade` or `--latest` either, has there been any actual reasons to require that when building the image? Otherwise better to keep it simple:

```suggestion
RUN apk add --no-cache build-base git gcc bash
```